engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/selectModulesScreen/SelectModulesScreen.java

```diff
@@ -106,6 +106,7 @@ public void onOpened() {
         }
 
         refreshSelection();
+        filterModules();
```

Comment:
This one fixes the problem with the list updating before the remote modules have downloaded.